docs/read-the-docs/source/connecting_nodes.rst

```diff
@@ -0,0 +1,83 @@
+Connecting nodes
+================
+
+Connect nodes via receptor backends. TCP, UDP, and websockets are currently supported. For example, ``tcp-peer`` can be used to connect to another node's ``tcp-listener``, and ``ws-peer`` can be used to connect to another node's ``ws-listener``.
+
+.. image:: mesh.png
+
+foo.yml
+
+.. code-block:: yaml
+
+    ---
+    - node:
+        id: foo
+
+    - log-level:
+        level: Debug
+
+    - tcp-listener:
+        port: 2222
+
+bar.yml
+
+.. code-block:: yaml
+
+    ---
+    - node:
+        id: bar
+
+    - log-level:
+        level: Debug
+
+    - tcp-peer:
+        address: localhost:2222
+
+fish.yml
+
+.. code-block:: yaml
+
+    ---
+    - node:
+        id: fish
+
+    - log-level:
+        level: Debug
+
+    - tcp-peer:
+        address: localhost:2222
+
+This forms a three-node mesh. Notice `bar` and `fish` are not directly connected to each other. However, the mesh allows traffic from `bar` to pass through `foo` to reach `fish`, as if `bar` and `fish` were directly connected.
+
+Logs from `fish` shows a successful connection to `bar` via `foo`.
+
+.. code-block::
+
+    INFO 2021/07/22 23:04:31 Known Connections:
+    INFO 2021/07/22 23:04:31    fish: foo(1.00)
+    INFO 2021/07/22 23:04:31    foo: bar(1.00) fish(1.00)
+    INFO 2021/07/22 23:04:31    bar: foo(1.00)
+    INFO 2021/07/22 23:04:31 Routing Table:
+    INFO 2021/07/22 23:04:31    foo via foo
+    INFO 2021/07/22 23:04:31    bar via foo
+
+
+Configuring backends
+^^^^^^^^^^^^^^^^^^^^
+
+``redial`` If set to true, receptor will automatically attempt to redial and restore connections that are lost.
+
+``cost``  User-defined metric that will be used by the mesh routing algorithm. If the mesh were represented by a graph node, then cost would be the length or weight of the edges between nodes. When the routing algorithm determines how to pass network packets from one node to another, it will use this cost to determine an efficient path.
+
+``nodecost`` Cost to a particular node on the mesh, and overrides whatever is set in ``cost``.
+
+.. code-block::
+
+    - tcp-listener:
+        port: 2222
+        cost: 1.0
+        nodecost:
```

Comment:
Can `nodecost` be used for `tcp-peer` as well? (Guessing not, might be good to clarify here)

Reply:
okay you make me realize I was missing an essential piece -- the cost must match on both ends of the connection. So for example fish's tcp-peer must have a cost of 2.0, to match the 2.0 on the tcp-listener in foo. I updated this part of the docs